### PR TITLE
segregating sites and Tajima's D

### PR DESCRIPTION
**Pull request by petrelharp**
Here's an implementation of "segregating sites"; see #230. Probably needs some C-level tests but is all good otherwise.

Next: [Tajima's D](https://en.wikipedia.org/wiki/Tajima%27s_D).

### CODE REVIEW

**Comment by petrelharp** [comment]
Screenshot from 2019-06-14 09-55-21

**Comment by petrelharp** [comment]
Darn it, how do I make my local builds as strict as AppVeyor's? I don't see any of these warnings when I compile.

**Comment by petrelharp** [comment]
Hm, issues seem to be in the Windows build. Anyone know what's going on with
```
======================================================================
ERROR: test_single_tree_jukes_cantor (tests.test_tree_stats.TestSiteDivergence)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\tskit\python\tests\test_tree_stats.py", line 655, in test_single_tree_jukes_cantor
    print(t.draw(format='unicode'))
  File "C:\Miniconda36-x64\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode characters in position 17-23: character maps to <undefined>
```

File: python/tests/tsutil.py [hunk 1]
```diff
@@ -333,8 +333,8 @@ def generate_site_mutations(tree, position, mu, site_table, mutation_table,
     more than one mutation per edge.
     """
     assert tree.interval[0] <= position < tree.interval[1]
-    states = {"A", "C", "G", "T"}
-    state = random.choice(list(states))
+    states = ["A", "C", "G", "T"]
+    state = random.choice(states)
```
Comment (petrelharp) [inline python/tests/tsutil.py]: The tests were not deterministic because of different orderings of sets, btw. This fixes that.

Comment (jeromekelleher) [reply python/tests/tsutil.py]: Dammit, got caught by that one before!

**Comment by jeromekelleher** [comment]
This is great! Can you add a low-level test also please like [this one](https://github.com/tskit-dev/tskit/blob/master/python/tests/test_lowlevel.py#L519)? This makes sure the Python C interface is properly exercised.

Happy to merge whenever.

**Comment by petrelharp** [comment]
I've added Tajima's D and rebased/squashed. Should be good to go; merge if you like (assuming it still passes tests...)

**Comment by jeromekelleher** [comment]
Looks great, merging!

**Pull request merged**